### PR TITLE
dev: modernjs devtools

### PR DESCRIPTION
**Pull request by Asuka109**
## Summary

Verify the process of basic development workflow and component injection.

image

### <samp>🤖 Generated by Copilot at 7e2b3bc</samp>

This pull request adds two new apps to the `web-infra-dev/modern.js` repository: a client app and a mount app. Both apps use the Modern.js framework and TypeScript, and have their own configuration files for browsers, ESLint, Git, npm, nvm, and tsconfig. The client app also has a basic router, head, and layout setup, and the mount app has a README file. Additionally, the pull request adds a heading to the `CONTRIBUTING.md` file of the `packages/devtools` directory.

### <samp>🤖 Generated by Copilot at 7e2b3bc</samp>

*  Add two apps, client and mount, to the devtools package using the Modern.js framework (F1-F19)

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/devtools/client/package.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    "@modern-js/builder-rspack-provider": "2.29.0",
+    "@modern-js/eslint-config": "2.29.0",
+    "@modern-js/plugin-proxy": "^2.29.0",
+    "@modern-js/tsconfig": "2.29.0",
+    "@types/jest": "~29.2.4",
+    "@types/lodash": "^4.14.196",
+    "@types/node": "~16.11.7",
+    "@types/react": "~18.0.26",
+    "@types/react-dom": "~18.0.10",
+    "devcert": "^1.2.2",
```
Comment (chenjiahan) [inline packages/devtools/client/package.json:36]: remove~

File: packages/devtools/client/package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "node": ">=16.18.1"
+  },
+  "dependencies": {
+    "@modern-js/runtime": "2.29.0",
+    "lodash": "^4.17.21",
+    "react": "~18.2.0",
+    "react-dom": "~18.2.0"
+  },
+  "devDependencies": {
+    "@modern-js-app/eslint-config": "2.29.0",
```
Comment (chenjiahan) [inline packages/devtools/client/package.json:25]: Use "workspace: *" for `@modern-js/xxx" packages

File: packages/devtools/client/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    "@modern-js/plugin-proxy": "^2.29.0",
+    "@modern-js/tsconfig": "2.29.0",
+    "@types/jest": "~29.2.4",
+    "@types/lodash": "^4.14.196",
+    "@types/node": "~16.11.7",
+    "@types/react": "~18.0.26",
+    "@types/react-dom": "~18.0.10",
+    "devcert": "^1.2.2",
+    "husky": "~8.0.1",
+    "lint-staged": "~13.1.0",
```
Comment (chenjiahan) [inline packages/devtools/client/package.json:38]: No need to install husky / lint-staged / prettier in sub-packages

File: packages/devtools/client/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    "dev": "modern dev",
+    "build": "modern build",
+    "start": "modern start",
+    "serve": "modern serve",
+    "new": "modern new",
+    "lint": "modern lint",
+    "upgrade": "modern upgrade"
+  },
+  "engines": {
+    "node": ">=16.18.1"
```
Comment (chenjiahan) [inline packages/devtools/client/package.json:16]: No 16.x

File: packages/devtools/client/package.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "@modern-js/devtools-client",
+  "version": "2.29.0",
+  "main": "./index.js",
+  "scripts": {
+    "reset": "npx rimraf ./**/node_modules",
+    "dev": "modern dev",
+    "build": "modern build",
```
Comment (chenjiahan) [inline packages/devtools/client/package.json:8]: modern-lib build

File: packages/devtools/mount/package.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    "node": ">=16.18.1"
+  },
+  "dependencies": {
+    "@modern-js/runtime": "2.29.0",
+    "react": "~18.2.0",
+    "react-dom": "~18.2.0",
+    "react-use": "^17.4.0"
+  },
+  "devDependencies": {
+    "@modern-js-app/eslint-config": "2.29.0",
```
Comment (chenjiahan) [inline packages/devtools/mount/package.json:25]: workspace: *

Comment (Asuka109) [reply packages/devtools/mount/package.json]: done.

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**